### PR TITLE
create workflow

### PR DESCRIPTION
**Pull request by guiltylotus**

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+          git tag --annotate --message "dmm-aggregator-backend $VERSION_TAG" "$VERSION_TAG"
+          git push origin "refs/tags/$VERSION_TAG"
+
+      - name: Create release
+        uses: softprops/action-gh-release@v1
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+          tag_name: ${{ env.VERSION_TAG }}
+          prerelease: false
+          name: "Pool Service ${{ env.VERSION_TAG }}"
```
Comment (lehainam-dev) [inline .github/workflows/release.yml:123]: This may need to be change.

Comment (guiltylotus) [reply .github/workflows/release.yml]: Updated ^^

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+  workflow_dispatch:
+    inputs:
+      version:
+        description: 'Release version'
+        type: string
+        required: true
+
+jobs:
+  prepare:
+    runs-on: [ self-hosted ]
```
Comment (it4rb) [inline .github/workflows/release.yml:13]:
```suggestion
    runs-on: [ ubuntu-22.04 ]
```
we cannot use self-hosted runner for public repo

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+      - name: Move Docker cache
+        run: |
+          rm -rf /tmp/.buildx-cache
+          mv /tmp/.buildx-cache-new /tmp/.buildx-cache
+
+  release:
+    needs:
+      - prepare
+      - docker
+    runs-on: [ self-hosted ]
```
Comment (it4rb) [inline .github/workflows/release.yml:92]:
```suggestion
    runs-on: [ ubuntu-22.04 ]
```

File: .github/workflows/release.yml [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+
+      - name: Setup Git
+        run: |
+          git config user.name "$GITHUB_ACTOR"
+          git config user.email "$GITHUB_ACTOR@users.noreply.github.com"
+
+      - name: Create tag
+        run: |
+          git tag -d "$VERSION_TAG" 2> /dev/null || echo "Release tag '$VERSION_TAG' does NOT exist"
+          git tag --annotate --message "dmm-aggregator-backend $VERSION_TAG" "$VERSION_TAG"
```
Comment (it4rb) [inline .github/workflows/release.yml:114]: um maybe we need to change this message?

File: .github/workflows/release.yml [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+        run: echo "::set-output name=value::$(date +%FT%T%z)"
+
+  docker:
+    needs:
+      - prepare
+    env:
+      SERVICE: kyberswap-dex-lib
+      VERSION_TAG: ${{ needs.prepare.outputs.version_tag }}
+      BUILD_DATE: ${{ needs.prepare.outputs.build_date }}
+    runs-on: [ self-hosted ]
```
[the remark this answers is not in the document]
Comment (guiltylotus) [reply .github/workflows/release.yml:38]: Why can not use `self-hosted` here?

File: .github/workflows/release.yml (hunk 4 above)
Comment (guiltylotus) [reply .github/workflows/release.yml]: Updated ^^

File: .github/workflows/release.yml (hunk 5 above)
[the remark this answers is not in the document]
Comment (it4rb) [reply .github/workflows/release.yml:38]: um this is a public repository, while our self-hosted runner is private, so github won't be able to run this action
other public repo need to use ubuntu as well, for example https://github.com/KyberNetwork/evmlistener/commit/285c591a611dbd4b38526528962a0f9853a71a76

**Review by it4rb — APPROVED** [review]

File: .github/workflows/release.yml (hunk 5 above)
[the remark this answers is not in the document]
Comment (guiltylotus) [reply .github/workflows/release.yml]: Updated

File: .github/workflows/release.yml (hunk 2 above)
Comment (guiltylotus) [reply .github/workflows/release.yml]: Updated

File: .github/workflows/release.yml (hunk 3 above)
Comment (guiltylotus) [reply .github/workflows/release.yml]: Updated

**Pull request merged**